test/memory/uninitialized_move.cpp

```diff
@@ -16,25 +16,29 @@
 #include <stl2/detail/algorithm/count_if.hpp>
 #include <stl2/detail/algorithm/equal.hpp>
 #include <stl2/detail/memory/destroy.hpp>
+#include <stl2/iterator.hpp>
 #include <stl2/view/repeat.hpp>
 #include <stl2/view/take_exactly.hpp>
+#include <algorithm>
 #include <cstdint>
 #include <memory>
 #include <vector>
 #include "../simple_test.hpp"
 #include "common.hpp"
 
 namespace ranges = __stl2;
+using ranges::ext::span;
 
 namespace {
-	template <typename T>
-	bool empty(const Array<T>& a) {
-		return ranges::all_of(a, &T::empty);
+	template <ranges::Range Rng>
+	bool empty(const Rng& rng, const std::ptrdiff_t n) {
+		return ranges::all_of(ranges::make_counted_iterator(rng.begin(), n), ranges::default_sentinel{},
+			&ranges::value_type_t<Rng>::empty);
 	}
 
-	template <typename T>
-	requires std::is_fundamental<T>::value
-	bool empty(const Array<T>&) {
+	template <ranges::Range Rng>
+	requires std::is_fundamental<ranges::value_type_t<Rng>>::value
```

Comment:
In product code I'd recommend requiring `typename ranges::value_type_t<Rng>` before using that type in a requirement like this (because sometimes better error messages) but this is test code so *shrug*.